mitodl/open-discussions
# Show paths and programs in search results

#### Pre-Flight checklist


#### What are the relevant tickets?
Partially addresses #2101

#### What's this PR do?
- Adds 'Program' and 'Learning Path' to search facets
- Enables search for the above
- Displays search results for the above
- Makes favoriting/unfavoriting work for the above

It does not display their details in the expanded drawer (clicking on the results won't do anything).

#### How should this be manually tested?
- Create some programs and learning paths via shell/notebook (I have a notebook I can send as an example).
- Recreate the ES index
- Go to `http://od.odl.local:8053/courses/search`.  You should see "Programs" and "Learning Paths" in the "Learning Resource" facet, and clicking on them should restrict search results to those types.  
- You should see the programs and learning paths in the results.  
- Favorite and unfavorite them, it should work the same as for courses and bootcamps.

The [design](https://projects.invisionapp.com/share/XRRX2PH5KZM#/screens/367253746) shows a price ("Free") and availability ("Available Now") for Programs which have neither attribute in the database (ditto Learning Paths) so I made those the default values to display if the search result doesn't have those attributes.

#### Screenshots (if appropriate)
![Screen Shot 2019-08-09 at 4 46 39 PM](https://user-images.githubusercontent.com/187676/62808159-4fb38700-bac5-11e9-99e5-c8bd49af58a4.png)
